reconcile/gitlab_owners.py

```diff
@@ -298,15 +302,52 @@ def format_report(report):
         return markdown_report.rstrip()
 
 
-def act(repo, dry_run, instance, settings):
+def generate_state_key(mr, comment):
+    user = comment['user']
+    comment_id = comment['id']
+    return f'{mr.target_project}-{mr.iid}-{user}-{comment_id}'
+
+
+# Each '/lgtm' must store the changes (diff) it is approving
+def store_diff_for_lgtm(dry_run, gl, mr, state):
+    comments = gl.get_merge_request_comments(mr.iid)
+    top_commit_created_at = gl.get_merge_request_top_commit_timestamp(mr)
+
+    for comment in comments:
+        # Only interested in '/lgtm' comments
+        if comment['body'] != '/lgtm':
+            continue
+
+        state_key = generate_state_key(mr, comment)
+        comment_created_at = dateparser.parse(comment['created_at'])
+
+        # Store diffs for only those comments that were created after the latest
+        # commit was pushed (the top commit)
+        # This was done in case, for some reason, the diff for an '/lgtm' is not
+        # immediately stored, during which it could be possible to accidentally add
+        # commits which have not been reviewed / approved
+        if comment_created_at < top_commit_created_at:
+            continue
+
```

Comment:
We should check if the key exists before trying to add it... it's cheaper (`self.exists(key)`)